source/interface/dav.js

```diff
@@ -19,7 +19,14 @@ function propsToStat(props, filename, isDetailed = false) {
         resourcetype: resourceType,
         getcontenttype: mimeType,
         getetag: etag
```

Comment:
Instead of setting them to a new object below, why not do something like this:

```javscript
const {
	getlastmodified: lastMod = null,
    getcontentlength: rawSize = "0",
    resourcetype: resourceType = null
   // ... etc
} = props;
```

That way you don't need to copy the entire object below.

Reply:
Good idea, I had no idea you could do something like that. But unfortunately that doesn't appear to work.

What I changed it to:
``` javascript
 16     const {
 17         getlastmodified: lastMod = null,
 18         getcontentlength: rawSize = "0",
 19         resourcetype: resourceType = null,
 20         getcontenttype: mimeType = null,
 21         getetag: etag = null
 22     } = props;
```

What gets generated:
``` javascript
 19   const _props$getlastmodifie = props.getlastmodified,
 20         lastMod = _props$getlastmodifie === void 0 ? null : _props$getlastmodifie,
 21         _props$getcontentleng = props.getcontentlength,
 22         rawSize = _props$getcontentleng === void 0 ? "0" : _props$getcontentleng,
 23         _props$resourcetype = props.resourcetype,
 24         resourceType = _props$resourcetype === void 0 ? null : _props$resourcetype,
 25         _props$getcontenttype = props.getcontenttype,
 26         mimeType = _props$getcontenttype === void 0 ? null : _props$getcontenttype,
 27         _props$getetag = props.getetag,
 28         etag = _props$getetag === void 0 ? null : _props$getetag;
 29   const type = resourceType && typeof resourceType === "object" && typeof resourceType.collection !== "undefined" ? "directory" : "file";
```